### PR TITLE
Add debug statements to istio-start.sh

### PR DESCRIPTION
**Pull request by MrTrustor**
This makes it easier to debug problems when trying to extend the mesh to a VM.

cc @costinm

### CODE REVIEW

File: tools/deb/istio-start.sh [hunk 1]
```diff
@@ -88,6 +88,16 @@ ISTIO_AGENT_FLAGS=${ISTIO_AGENT_FLAGS:-}
 # Split ISTIO_AGENT_FLAGS by spaces.
 IFS=' ' read -r -a ISTIO_AGENT_FLAGS_ARRAY <<< "$ISTIO_AGENT_FLAGS"
 
+# Dump out our environment for debugging purposes.
+echo "Environment:"
+echo "------------"
+echo "ISTIO_PILOT_PORT=${ISTIO_PILOT_PORT-}"
```
[the remark this answers is not in the document]
Comment (MrTrustor) [reply tools/deb/istio-start.sh]: Apparently not: https://stackoverflow.com/questions/32674713/what-does-the-dash-after-variable-names-do-here
I copied the syntax in [istio-iptables.sh](https://github.com/istio/istio/blob/master/tools/deb/istio-iptables.sh#L175), I'm nowhere good enough in bash to come up with that on my own 😄

**Comment by costinm** [comment]
Thanks for the PR - if you have extra time maybe you can also take a look at https://github.com/istio/istio/issues/9240 ( just add an env variable to the same script and test it, I'm a bit behind on my list ...)

**Comment by hklai** [comment]
master is still locked. Please rebase to release-1.1 for testing.

**Comment by costinm** [comment]
Can you rebase ? Or close it, doesn't seem critical for 1.1...

**Comment by ymesika** [comment]
Do you mind switching to `master`?

**Comment by MrTrustor** [comment]
@costinm @ymesika Rebased and switched to master.
Can we merge this?

**Pull request closed without merging**